python_modules/dagster-externals/dagster_externals/_protocol.py

```diff
@@ -1,20 +1,36 @@
 from typing import Any, Mapping, Optional, Sequence
 
-from typing_extensions import Final, TypeAlias, TypedDict
+from typing_extensions import Final, NotRequired, TypeAlias, TypedDict
 
 ExternalExecutionExtras: TypeAlias = Mapping[str, Any]
 
+ENV_KEY_PREFIX: Final = "DAGSTER_EXTERNALS_"
+
+
+def _param_name_to_env_key(key: str) -> str:
+    return f"{ENV_KEY_PREFIX}{key.upper()}"
+
+
 # ##### PARAMETERS
 
 DAGSTER_EXTERNALS_DEFAULT_INPUT_FILENAME: Final = "dagster_external_input"
 DAGSTER_EXTERNALS_DEFAULT_OUTPUT_FILENAME: Final = "dagster_external_output"
 
 DAGSTER_EXTERNALS_ENV_KEYS: Final = {
-    "is_orchestration_active": "DAGSTER_EXTERNALS_IS_ORCHESTRATION_ACTIVE",
-    "input": "DAGSTER_EXTERNALS_INPUT",
-    "output": "DAGSTER_EXTERNALS_OUTPUT",
+    k: _param_name_to_env_key(k)
+    for k in ("is_orchestration_active", "input_reader", "output_writer")
 }
 
+
+# ##### IO
+
+
+class IOSpec(TypedDict):
+    module: NotRequired[Optional[str]]
+    attribute: NotRequired[Optional[str]]
+    params: NotRequired[Optional[Mapping[str, Any]]]
```

Comment:
not immeadiately obvious what this is for. Name very generic and collides with AssetSpec et al